src/arch/svm/SpokeUtils.ts

```diff
@@ -113,17 +116,73 @@ export async function getSlotForBlock(
   return undefined;
 }
 
-export function findDepositBlock(
-  _spokePool: unknown,
-  depositId: BigNumber,
-  _lowBlock: number,
-  _highBlock?: number
-): Promise<number | undefined> {
+/**
+ * Finds the block number and slot for a given deposit ID by querying for deposit events
+ * within a limited time range.
+ *
+ * @remarks
+ * This implementation uses a time-limited search approach because Solana PDA state has
+ * limitations that prevent directly referencing old deposit IDs. Unlike EVM chains where
+ * we might use binary search across the entire chain history, in Solana we must query within
+ * a constrained block range.
+ *
+ * We use a 2-day lookback window as a heuristic because:
+ * 1. Most valid deposits that need to be processed will be recent
+ * 2. This covers multiple bundle submission periods
+ * 3. It balances performance with practical deposit age
+ *
+ * @important
+ * This function may return `undefined` for valid deposit IDs that are older than the search
+ * window (approximately 2 days). This is an acceptable limitation as deposits this old are
+ * typically not relevant to current operations.
+ *
+ * @param provider - Solana RPC provider
+ * @param depositId - The deposit ID to search for
+ * @returns The deposit, block number, and slot where the deposit occurred, or undefined if not found
+ */
+export async function findDeposit(
+  provider: Provider,
+  depositId: BigNumber
+): Promise<{ block: BigNumber; slot: BigNumber; deposit: DepositWithBlock } | undefined> {
```

Comment:
We're already returning `DepositWithBlock`; I assume that'll now contain a slot, but is there any reason to return `block` or `slot` separately? There's at least some duplication there.